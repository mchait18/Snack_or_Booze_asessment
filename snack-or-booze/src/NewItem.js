import React, { useState } from "react";
import "./Form.css"
import { Redirect } from "react-router-dom";
import { Card, CardBody, CardTitle, ListGroup, ListGroupItem } from "reactstrap";
import "./FoodMenu.css";

function NewItem({ addItem }) {
    const INITIAL_STATE = {
        name: '',
        description: "",
        recipe: '',
        serve: "",
        type: ""
    }
    const [formData, setFormData] = useState(INITIAL_STATE);
    const [formComplete, setFormComplete] = useState(false)
    const [url, setUrl] = useState("/")

    const handleChange = (e) => {
        const { name, value } = e.target;
        setFormData(formData => ({
            ...formData,
            [name]: value
        }))
    }
    const handleSubmit = (e) => {
        e.preventDefault();
        addItem({ ...formData });
        setFormData(INITIAL_STATE)
        //set form as complete and set url to snacks or drinks
        setFormComplete(true)
        setUrl(`/${formData.type}`)
    }
    //if form is submitted, redirect to either snacks or drinks page. Otherwise, load form
    return (
        <section>
            {formComplete && <Redirect to={url} />}
            <Card>
                <CardBody>
                    <CardTitle className="font-weight-bold text-center">
                        Add a New Snack or Drink
                    </CardTitle>
                    <ListGroup>
                        <form onSubmit={handleSubmit}>
                            <ListGroupItem>
                                <fieldset>
                                    <div>
                                        <input
                                            id="snack"
                                            type="radio"
                                            name="type"
                                            value="snacks"
                                            onChange={handleChange}
                                        />
                                        <label htmlFor="snack">Snack</label>
                                    </div>
                                    <div>
                                        <input
                                            id="drink"
                                            type="radio"
                                            name="type"
                                            value="drinks"
                                            onChange={handleChange}
                                        />
                                        <label htmlFor="drink">Drink</label>
                                    </div>
                                </fieldset>
                            </ListGroupItem>
                            <ListGroupItem>
                                <label htmlFor="name">Name: </label>
                                <input
                                    style={{ width: "310px" }}
                                    id="name"
                                    type="text"
                                    name="name"
                                    value={formData.name}
                                    onChange={handleChange}
                                />
                            </ListGroupItem>
                            <ListGroupItem>
                                <label htmlFor="name">Description: </label>
                                <input
                                    style={{ width: "265px" }}
                                    id="description"
                                    type="text"
                                    name="description"
                                    value={formData.description}
                                    onChange={handleChange}
                                />
                            </ListGroupItem>
                            <ListGroupItem>
                                <label htmlFor="val">Recipe: </label>
                                <textarea
                                    cols="38"
                                    id="recipe"
                                    name="recipe"
                                    value={formData.recipe}
                                    onChange={handleChange}
                                />
                            </ListGroupItem>
                            <ListGroupItem>
                                <label htmlFor="val">Serve: </label>
                                <input
                                    style={{ width: "310px" }}
                                    id="serve"
                                    type="text"
                                    name="serve"
                                    value={formData.serve}
                                    onChange={handleChange}
                                />
                            </ListGroupItem>
                            <button>Add Item</button>
                        </form >
                    </ListGroup>
                </CardBody>
            </Card>
        </section>
    )
}

export default NewItem