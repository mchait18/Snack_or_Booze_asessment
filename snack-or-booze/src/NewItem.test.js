import React from "react";
import { render, fireEvent } from "@testing-library/react";
import NewItem from "./NewItem";

function addItem(item, itemName = "Apples") {
    const itemInput = item.getByLabelText("Name:")
    fireEvent.change(itemInput, { target: { value: itemName } })
    const addButton = item.getByText("Add Item")
    fireEvent.click(addButton)
}

it("renders without crashing", function () {
    render(<NewItem />);
});

it("matches snapshot", function () {
    const { asFragment } = render(<NewItem />);
    expect(asFragment()).toMatchSnapshot();
});


it("can add an item", function () {
    const item = render(<NewItem />);
    // addItem(item);
    // expect form to clear and todo to be on the page
    // expect(item.getByLabelText("Task:")).toHaveValue("");
    // expect(list.getByText("write tests")).toBeInTheDocument();
    // expect(list.getByText("Apples")).toBeInTheDocument();
    // expect(list.getByText("Add Item")).not.toBeInTheDocument();
});  