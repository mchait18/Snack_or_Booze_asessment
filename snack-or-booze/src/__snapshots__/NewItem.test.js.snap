// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <section>
    <div
      class="card"
    >
      <div
        class="card-body"
      >
        <h5
          class="font-weight-bold text-center card-title"
        >
          Add a New Snack or Drink
        </h5>
        <ul
          class="list-group"
        >
          <form>
            <li
              class="list-group-item"
            >
              <fieldset>
                <div>
                  <input
                    id="snack"
                    name="type"
                    type="radio"
                    value="snacks"
                  />
                  <label
                    for="snack"
                  >
                    Snack
                  </label>
                </div>
                <div>
                  <input
                    id="drink"
                    name="type"
                    type="radio"
                    value="drinks"
                  />
                  <label
                    for="drink"
                  >
                    Drink
                  </label>
                </div>
              </fieldset>
            </li>
            <li
              class="list-group-item"
            >
              <label
                for="name"
              >
                Name: 
              </label>
              <input
                id="name"
                name="name"
                style="width: 310px;"
                type="text"
                value=""
              />
            </li>
            <li
              class="list-group-item"
            >
              <label
                for="name"
              >
                Description: 
              </label>
              <input
                id="description"
                name="description"
                style="width: 265px;"
                type="text"
                value=""
              />
            </li>
            <li
              class="list-group-item"
            >
              <label
                for="val"
              >
                Recipe: 
              </label>
              <textarea
                cols="38"
                id="recipe"
                name="recipe"
              />
            </li>
            <li
              class="list-group-item"
            >
              <label
                for="val"
              >
                Serve: 
              </label>
              <input
                id="serve"
                name="serve"
                style="width: 310px;"
                type="text"
                value=""
              />
            </li>
            <button>
              Add Item
            </button>
          </form>
        </ul>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
